src/realm/integer_flex_compressor.cpp

```diff
@@ -70,10 +70,17 @@ void FlexCompressor::copy_data(const Array& arr, const std::vector<int64_t>& val
 bool FlexCompressor::find_all_match(size_t start, size_t end, size_t baseindex, QueryStateBase* state)
 {
     REALM_ASSERT_DEBUG(state->match_count() < state->limit());
-    const auto process = state->limit() - state->match_count();
-    const auto end2 = end - start > process ? start + process : end;
-    for (; start < end2; start++)
-        if (!state->match(start + baseindex))
+    while (start < end) {
+        if (!state->match(start++ + baseindex))
```

Comment:
NIT: This is the only loop in this file in which we use post increment and + operator together. Maybe align this to the other loops.

Reply:
👍